src/app/recipient/components/recipient-earned-badge-list/recipient-earned-badge-list.component.html

```diff
@@ -1,255 +1,220 @@
-									>
-										{{ issuerGroup.issuer.name }}
-									</a>
-								</th>
-							</tr>
-							<!-- Badges -->
-							<ng-template ngFor let-badge [ngForOf]="issuerGroup.badges" let-i="index">
-								<tr class="datatable-x-row">
-									<td class="datatable-x-cell">
-										<div class="l-flex l-flex-2x l-flex-aligncenter">
-											<img
-												class="badgeimage badgeimage-{{
-													badge.mostRelevantStatus
-												}} badgeimage-small u-margin-right2x l-flex-x-shrink0"
-												width="40"
-												[loading-src]="badgeLoadingImageUrl"
-												[error-src]="badgeFailedImageUrl"
-												[loaded-src]="badge.image"
-											/>
-											<a
-												class="u-text u-text-link u-text-breakword"
-												[routerLink]="['../earned-badge', badge.slug]"
-												>{{ badge.badgeClass.name }}</a
-											>
-											<div
-												class="datatable-x-status badgestatus badgestatus-intable badgestatus-{{
-													badge.mostRelevantStatus
-												}}"
-												*ngIf="badge.mostRelevantStatus"
-											>
-												{{ badge.mostRelevantStatus }}
-											</div>
-										</div>
-									</td>
-									<td class="datatable-x-cell u-text-body">
-										<a
-											class="u-text u-text-link u-text-breakword"
-											[routerLink]="[
-												'../../public/issuers',
-												issuerIdToSlug(issuerGroup.issuerId)
-											]"
-										>
+						<!-- No badges, empty state! -->
+						<section class="emptysection l-emptysection" *ngIf="allBadges.length == 0">
+							<h2 class="emptysection-x-headline">{{ 'RecBadge.noBadgesYet' | translate }}</h2>
+							<p class="emptysection-x-copy">{{ 'RecBadge.collectShareBadges' | translate }}</p>
+							<div class="emptysection-x-imagecontainer">
+								<img alt="Illustration of backpack with badges" class="emptysection-x-image"
+									[src]="noBadgesImageUrl" />
+							</div>
+						</section>
+
+						<!-- Badge Search, group and grid/list toggle -->
+						<div class="oeb-section-sm" *ngIf="allBadges.length > 0">
+							<div class="tw-flex tw-flex-wrap tw-items-center oeb-standard-padding-bottom">
+								<input type="text" name="forminput" id="forminput" placeholder="Badges durchsuchen"
+									[(ngModel)]="searchQuery"
+									class="tw-w-[300px] tw-border-solid tw-border-purple tw-bg-white tw-mr-6"
+									hlmInput />
+								<oeb-checkbox class="tw-mt-2 md:tw-mt-0" name="groupby" [(ngModel)]="groupByIssuer"
+									[text]="'RecBadge.groupByInstitution' | translate"></oeb-checkbox>
+							</div>
+
+							<!-- Grid View -->
+							<ng-template [ngIf]="badgesDisplay == 'grid'">
+								<!-- ------------------ UNGROUPED ------------------  -->
+								<ng-template [ngIf]="!groupByIssuer">
+									<div class="l-grid l-grid-large tw-gap-10">
+										<!-- TODO: Fix the badgecard link for sharing and routing -->
```

Comment:
Does this TODO already have a ticket?

Reply:
Was an old comment. Removed it. Will clarify rooting with sven.